diff --git a/buildtools/wafsamba/samba_conftests.py b/buildtools/wafsamba/samba_conftests.py
index c0b9ae4..5b91451 100644
--- a/buildtools/wafsamba/samba_conftests.py
+++ b/buildtools/wafsamba/samba_conftests.py
@@ -81,50 +81,6 @@ def CHECK_ICONV(conf, define='HAVE_NATIVE_ICONV'):
     return False
 
 
-@conf
-def CHECK_LARGEFILE(conf, define='HAVE_LARGEFILE'):
-    '''see what we need for largefile support'''
-    getconf_cflags = conf.CHECK_COMMAND(['getconf', 'LFS_CFLAGS']);
-    if getconf_cflags is not False:
-        if (conf.CHECK_CODE('return !(sizeof(off_t) >= 8)',
-                            define='WORKING_GETCONF_LFS_CFLAGS',
-                            execute=True,
-                            cflags=getconf_cflags,
-                            msg='Checking getconf large file support flags work')):
-            conf.ADD_CFLAGS(getconf_cflags)
-            getconf_cflags_list=TO_LIST(getconf_cflags)
-            for flag in getconf_cflags_list:
-                if flag[:2] == "-D":
-                    flag_split = flag[2:].split('=')
-                    if len(flag_split) == 1:
-                        conf.DEFINE(flag_split[0], '1')
-                    else:
-                        conf.DEFINE(flag_split[0], flag_split[1])
-
-    if conf.CHECK_CODE('return !(sizeof(off_t) >= 8)',
-                       define,
-                       execute=True,
-                       msg='Checking for large file support without additional flags'):
-        return True
-
-    if conf.CHECK_CODE('return !(sizeof(off_t) >= 8)',
-                       define,
-                       execute=True,
-                       cflags='-D_FILE_OFFSET_BITS=64',
-                       msg='Checking for -D_FILE_OFFSET_BITS=64'):
-        conf.DEFINE('_FILE_OFFSET_BITS', 64)
-        return True
-
-    if conf.CHECK_CODE('return !(sizeof(off_t) >= 8)',
-                       define,
-                       execute=True,
-                       cflags='-D_LARGE_FILES',
-                       msg='Checking for -D_LARGE_FILES'):
-        conf.DEFINE('_LARGE_FILES', 1)
-        return True
-    return False
-
-
 @conf
 def CHECK_C_PROTOTYPE(conf, function, prototype, define, headers=None, msg=None):
     '''verify that a C prototype matches the one on the current system'''
diff --git a/buildtools/wafsamba/wscript b/buildtools/wafsamba/wscript
index 7b8fb01..f82f283 100644
--- a/buildtools/wafsamba/wscript
+++ b/buildtools/wafsamba/wscript
@@ -477,10 +477,6 @@ struct foo bar = { .y = 'X', .x = 1 };
     conf.CHECK_HEADERS('stdbool.h stdint.h stdarg.h vararg.h', add_headers=True)
     conf.CHECK_HEADERS('limits.h assert.h')
 
-    # see if we need special largefile flags
-    if not conf.CHECK_LARGEFILE():
-        raise Errors.WafError('Samba requires large file support support, but not available on this platform: sizeof(off_t) < 8')
-
     if conf.env.HAVE_STDDEF_H and conf.env.HAVE_STDLIB_H:
         conf.DEFINE('STDC_HEADERS', 1)
 
